clang/lib/Basic/DiagnosticIDs.cpp

```diff
@@ -489,13 +485,7 @@ static DiagnosticIDs::Level toLevel(diag::Severity SV) {
 DiagnosticIDs::Level
 DiagnosticIDs::getDiagnosticLevel(unsigned DiagID, SourceLocation Loc,
                                   const DiagnosticsEngine &Diag) const {
-  // Handle custom diagnostics, which cannot be mapped.
```

Comment:
so despite this change making sense at a high level, this is a big behavior change that isn't really obvious in context of this PR.

before this change, clang didn't propagate `-Werror` like mappings into custom diagnostics. Hence some compilations that succeed now with custom warning (clang-plugins etc.), will start failing after this change as those will now turn into errors.
We're already seeing this internally and both https://github.com/llvm/llvm-project/pull/70976#issuecomment-2357811301 and https://github.com/llvm/llvm-project/pull/70976#issuecomment-2372463357 are mentioning this as well.

Since this is triggering a backward incompatible behavior, can we have this bit reverted and decide how/whether we're planning to move forward with this?

Reply:
> Since this is triggering a backward incompatible behavior, can we have this bit reverted and decide how/whether we're planning to move forward with this?

Afaict the author ( @philnik777 ) has neither responded to the post-commit comments/question here, neither in the PR related to the reapply at https://github.com/llvm/llvm-project/pull/108453#issuecomment-2352155766 , and I don't know much about this to understand if there is an easy fix. But if we are suspecting "multiple bugs here" then maybe someone should just go ahead and revert this again.

Btw, it seems like the reapply already has been reverted once (https://github.com/llvm/llvm-project/commit/e1bd9740faa62c11cc785a7b70ec1ad17e286bd1) and then reapplied a second time (https://github.com/llvm/llvm-project/commit/e39205654dc11c50bd117e8ccac243a641ebd71f). And there are comments here https://github.com/llvm/llvm-project/commit/e39205654dc11c50bd117e8ccac243a641ebd71f#commitcomment-146720911 about further fixups that might cause conflicts if reverting again.

PS. The commit message in https://github.com/llvm/llvm-project/commit/e39205654dc11c50bd117e8ccac243a641ebd71f is not very informative as it says that the commit msg body says that it is a revert, while it actually is a re-apply. Maybe that is another reason to revert this mess and reapply it using a new PR with an appropriate commit message describing the feature.

Reply:
I agree, let's revert if this is causing problems.

Reply:
ok, putting together a revert for https://github.com/llvm/llvm-project/commit/e39205654dc11c50bd117e8ccac243a641ebd71f